app.js

```diff
@@ -1,140 +1,65 @@
-      album: (res.track.album_resource ? res.track.album_resource.name : ""),
-      artist: (res.track.artist_resource ? res.track.artist_resource.name : ""),
-      playing: res.playing,
-      position: res.playing_position,
-      length: res.track.length,
-      start,
-      end
-    };
-    
-    currentSong = song;
-
-    songEmitter.emit('newSong', song);
-  });
-}
+try {
+	const path = process.platform === 'win32' ? 'C:\\Windows\\System32\\drivers\\etc\\hosts' : '/hosts/etc';
+	const file = fs.readFileSync(path);
+	if (file.includes('open.spotify.com')) {
+		log(`Arr' yer be pirating! Please remove your "open.spotify.com" rule from your hosts file located in ${path}`);
+		process.exit(1);
+	}
+} catch (error) {} // eslint-disable-line no-empty
+
+const activity = {
+	details: '🎵  Nothing',
+	largeImageKey: cfg.images.large,
+	smallImageKey: cfg.images.smallPaused
+};
 
 /**
- * Initialise song listeners
- * newSong: gets emitted when the song changes to update the RP
- * songUpdate: currently gets executed when the song gets paused/resumes playing.
+ * Helper function to change the activity data
  **/
-songEmitter.on('newSong', song => {
-  rpc.setActivity({
-    details: `🎵  ${song.name}`,
-    state: `👤  ${song.artist}`,
-    startTimestamp: song.start,
-    endTimestamp: song.end,
-    largeImageKey,
-    smallImageKey,
-    largeImageText: `⛓  ${song.uri}`,
-    smallImageText: `💿  ${song.album}`,
-    instance: false,
-  });
-
-  log(`Updated song to: ${song.artist} - ${song.name}`);
-});
-
-songEmitter.on('songUpdate', song => {
-  let startTimestamp = song.playing ?
-    parseInt(new Date().getTime().toString().substr(0, 10)) - song.position :
-    undefined,
-    endTimestamp = song.playing ?
-    startTimestamp + song.length :
-    undefined;
-
-  rpc.setActivity({
-    details: `🎵  ${song.name}`,
-    state: `👤  ${song.artist}`,
-    startTimestamp,
-    endTimestamp,
-    largeImageKey,
-    smallImageKey: startTimestamp ? smallImageKey : smallImagePausedKey,
-    largeImageText: `⛓  ${song.uri}`,
-    smallImageText: `💿  ${song.album}`,
-    instance: false,
-  });
+function update(song, image = cfg.images.smallPaused) {
+	activity.smallImageKey = image;
+	if (song) {
+		activity.details = `🎵  ${song.title}`;
+		activity.state = `👤  ${song.artist.name}`;
+		activity.largeImageText = `🔗  ${song.id}`;
+		activity.smallImageText = `💿  ${song.album.id}`;
+		activity.startTimestamp = Math.floor(Date.now() / 1000) - song.played;
+		activity.endTimestamp = activity.startTimestamp + song.length;
+	} else {
+		activity.details = '🎵  Nothing';
+		delete activity.state;
+		delete activity.smallImageText;
+		delete activity.largeImageText;
+		delete activity.startTimestamp;
+		delete activity.endTimestamp;
+	}
+}
 
-  log(`Song state updated (playing: ${song.playing})`)
+client.on('ready', () => {
+	log(`Connected to Discord! (${cfg.id})`);
+	setInterval(() => client.setActivity(activity), 15e3);
```

Comment:
15e3?

Reply:
15e3 = 15000 = 15 * 10 ^ 3

Reply:
What if the user changes song? It would create quite a noticeable gap.